recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,218 @@
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
+        tc.variables['BUILD_TESTING'] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+        tc.variables["STLAB_USE_BOOST_CPP17_SHIMS"] = self.options.with_boost
+        tc.variables["STLAB_NO_STD_COROUTINES"] = self.options.no_std_coroutines
+        tc.variables["STLAB_THREAD_SYSTEM"] = self.options.thread_system
+        tc.variables["STLAB_TASK_SYSTEM"] = self.options.task_system
+
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        tc.generate()
+
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self.copy("*LICENSE", dst="licenses", keep_path=False)
-        self.copy("stlab/*", src=self._source_subfolder, dst='include/')
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_id(self):
-        self.info.header_only()
-        self.info.options.boost_optional = "ANY"
-        self.info.options.boost_variant = "ANY"
+        #TODO: is header only but needs a header modified by cmake
+        # self.info.settings.clear()
+        # self.info.header_only()
+        pass
 
     def package_info(self):
-        coroutines_value = 1 if self.options.coroutines else 0
+        future_coroutines_value = 1 if self.options.future_coroutines else 0
 
         self.cpp_info.defines = [
-            'STLAB_FUTURE_COROUTINES={}'.format(coroutines_value)
+            'STLAB_FUTURE_COROUTINES={}'.format(future_coroutines_value)
         ]
 
-        if self.options.boost_optional:
-            self.cpp_info.defines.append("STLAB_FORCE_BOOST_OPTIONAL")
-
-        if self.options.boost_variant:
-            self.cpp_info.defines.append("STLAB_FORCE_BOOST_VARIANT")
-
-        if self.options.task_system == "portable":
-            self.cpp_info.defines.append("STLAB_FORCE_TASK_SYSTEM_PORTABLE")
-        elif self.options.task_system == "libdispatch":
-            self.cpp_info.defines.append("STLAB_FORCE_TASK_SYSTEM_LIBDISPATCH")
-        elif self.options.task_system == "emscripten":
-            self.cpp_info.defines.append("STLAB_FORCE_TASK_SYSTEM_EMSRIPTEN")  #Note: there is a typo in Stlab Cmake.
-            self.cpp_info.defines.append("STLAB_FORCE_TASK_SYSTEM_EMSCRIPTEN") #Note: for typo fix in later versions
-        elif self.options.task_system == "pnacl":
-            self.cpp_info.defines.append("STLAB_FORCE_TASK_SYSTEM_PNACL")
-        elif self.options.task_system == "windows":
-            self.cpp_info.defines.append("STLAB_FORCE_TASK_SYSTEM_WINDOWS")
+        if self.settings.os == "Windows":
+            self.cpp_info.defines = ['NOMINMAX']
```

Comment:
Consumer should be settings this 

```suggestion
```

Reply:
If I remove those lines I get the following error:

https://c3i.jfrog.io/c3i/misc/logs/pr/13048/67-configs/windows-visual_studio/stlab/1.7.1//02833e547ecc88a0495cb145970002b7a2796825-test.txt

I do not like the idea that the consumer recipe has to know about these internal things...